### PR TITLE
Fix for mobile scrollable tables

### LINKED ISSUE #31: Scrollable tables don't show content on mobile
Antd does not have an easy API to have tables to stretch to their parent container, so the css has to be modified for antd-tables. Making the table scrollable makes it worse, since there is a hacky way to make scrolling in a stretchable table work i.e. setting the height of the table-body to `0px` even though it is definitely not `0px`. This works on desktop browsers, but not for mobile browsers. The only solution would be to have a fixed height, but that doesn't work well as being responsive.

### PR DESCRIPTION
**Pull request by walterimaican**
Resolves #31 

Root cause analysis revealed that due to conflicting CSS properties on ANTD elements, the table was not rendering properly on mobile devices. 

Removing the conflicting CSS yielded a table that spilled out over its parent container. As a result, this pull request includes a hacky solution to dynamically bounding the expenses table within its ancestor elements.

A better long-term solution would involve taking another look at the CSS layout and architecture of the frontend and perhaps overhauling it in the future (see [#48](https://github.com/jfabellera/billy/issues/48)).

### CODE REVIEW

**Comment by jfabellera** [comment]
Before testing it out on mobile I wanted to make sure everything looked fine as before on desktop, but I found the following issues:
- on /dashboard
   - Table causes overflow for the entire page (~15px). I would like it stretch to its parent element like before.
   - Table is missing pagination footer.
- on /expenses
   - Table has no internal overflow i,e, table is as tall as the number of list items it contains, resulting in a large amount of overall page overflow.

And on mobile I found:
- on /dashboard
   - Table shows expenses but is not scrollable.
   - Table is missing pagination footer.
- on /expenses
   - Same as desktop problem

Ran this both on my local machine and a development server, and the results were consistent.

**Comment by walterimaican** [comment]
[Line 180](https://github.com/jfabellera/billy/pull/49/files#diff-b4d72996ed83aaf474faa02bb3c71c4e0c06182e5bf7fd7721160a509d741348R180) removes the pagination if there is only one page - I added this line while experimenting and forgot to remove it

I'll look further into the issues mentioned

**Comment by walterimaican** [comment]
Desktop
- dashboard:
  - [x] overflow: Manually adjusted ratio so that there isn't a spillover. Having it stretch to height of parent element is the preferred method, however the existing CSS breaks mobile use. This change is a compromise to see on mobile.
  - [x] pagination footer: Added back (had it set so that if there is only one page, no pagination footer needed)
- expenses
  - [x] overflow: corrected

Mobile
- dashboard:
  - [x] scrollable: Table is actually scrollable, it just does not show the scrollbar - limitation with antd table integration
  - [x] pagination footer: fixed, see above
- expenses:
  - [x] overflow: corrected

File: frontend/src/containers/expenses/dashboard.jsx [hunk 1]
```diff
@@ -88,23 +116,26 @@ class Dashboard extends Component {
...
-              style={{ height: '100%' }}
             >
               <ExpensesTable
                 title={moment(new Date(this.state.start_date)).format('MMMM')}
                 options={{
                   start_date: this.state.start_date,
                   end_date: this.state.end_date,
                 }}
+                tableHeight={
+                  this.state.dashboardHeight - this.state.topRowHeight
```
Comment (jfabellera) [inline frontend/src/containers/expenses/dashboard.jsx]: Should handle when the ref doesnt have a height yet i.e. before component is mounted. Getting a warning for `maxHeight: NaN`

e.g.
```javascript
(this.state.dashboardHeight - this.state.topRowHeight) || 0
```

**Review by jfabellera — CHANGES_REQUESTED** [review]

File: frontend/src/containers/expenses/dashboard.jsx (hunk 1 above)
Comment (walterimaican) [reply frontend/src/containers/expenses/dashboard.jsx]: Good catch, added default value in commit d6934e3

**Comment by jfabellera** [comment]
There are some styling issues, but they can be sorted out in another PR and I don't think are a result of this current PR. So I am gonna go ahead and merge this.

**Pull request merged**